### ISSUE TITLE
Dont allow creation of file without a name instead of falling back to gistfile*.txt

### ISSUE THREAD

**Issue by pavelloz** [enjalot/blockbuilder#161]
Currently its possible to create a file without name, which is pretty weird.

Its even more surprising after you save the block, then (github?) its name changes into gistfile<number>.txt -- which is better than error, but maybe it would be better experience if filename had to be filled by user.

**Comment by enjalot** [comment]
yeah, we shouldn't allow a file without a name. I'd also like to support renaming files at some point...
